deluge/httpdownloader.py

```diff
@@ -328,3 +381,62 @@ def on_download_fail(failure):
     )
     d.addCallbacks(on_download_success, on_download_fail)
     return d
+
+
+def download_file_to_mem(
```

Comment:
Can we move this changeset to a separate PR? Also no sure about the naming of this method but will talk about that in new PR

Reply:
Will do